File: Solved-Whiteboard-Challenges/RingBuffer/RingBuffer.js



class RingBuffer {
    constructor(max) {
      this.max = max;
      this.pointer = 0;
      this.storage = []
      for (let i = 1; i <= this.max; i++) {
          this.storage.push(null);
      }
    }

    insert(value) {
        this.storage[this.pointer] = value;
        this.pointer += 1;
        if (this.pointer >= this.max) {
            this.pointer = 0;
        }
    }

    get() {
        console.log(this.storage);
    }
}


const x = new RingBuffer(3);

x.insert('a');
x.insert('b');
x.insert('c');

x.get();

x.insert('d');
x.get();
x.insert('e');
x.get();
x.insert('f');
x.get();

File: Solved-Whiteboard-Challenges/RingBuffer/ring_buffer.py


class RingBuffer:
    def __init__(self, limit):
        self.limit = limit
        self.storage = [None] * limit 
        self.pointer = 0
    
    def append(self, item):
        self.storage[self.pointer] = item
        self.pointer += 1
        self.pointer = self.pointer % self.limit

    def allValues(self):
        print(self.storage)


buffer = RingBuffer(3)

buffer.append('a')
buffer.append('b')
buffer.append('c')

buffer.allValues()   # should return ['a', 'b', 'c']

buffer.append('d')

buffer.allValues()   # should return ['d', 'b', 'c']

buffer.append('e')
buffer.append('f')

buffer.allValues()   # should return ['d', 'e', 'f']
